src/main/java/entities/hazards/MovingEnemy.java

```diff
@@ -0,0 +1,55 @@
+package entities.hazards;
+
+/** An enemy which moves. */
+public abstract class MovingEnemy extends Enemy {
+    /**
+     * Current x position of enemy
+     */
+    private int currX;
+    /**
+     * Current y position of enemy
+     */
+    private int currY;
+
```

Comment:
Maybe you have a different idea, but wouldn't the enemy need a direction variable too?

Reply:
Only if he wants to design 4 different sprites for 4 directions. I think it's optional for our discrete game. 
For "continuous" game, the direction might becomes important tho.